QMLComponents/analysisform.cpp

```diff
@@ -660,7 +660,7 @@ void AnalysisForm::setAnalysisUp()
 	_initialized = true;
 
 	// Don't bind boundValuesChanged before it is initialized: each setup of all controls will generate a boundValuesChanged
-	connect(_analysis,					&AnalysisBase::boundValuesChanged,		this,			&AnalysisForm::setRSyntaxText,				Qt::QueuedConnection	);
+	connect(_analysis,					&AnalysisBase::boundValuesChanged,		this,			&AnalysisForm::rSyntaxTextChanged,				Qt::QueuedConnection	);
```

Comment:
So when an option changes we get a analysis to run and also immediately the rsyntax rscript?

Cant the analysis just return the syntax as part of its results or something like that?
This just seems like asking for trouble (race conditions, weird engine kills etc)

Reply:
I think the R Syntax window should rather reflect the state of the options. If the options give an analysis error, the R Syntax should still display these faulty options, whatever the results of the analysis.
Also an analysis can run for quite long: there is no reason to wait for the end of the analysis to display the R Syntax. I don't see race conditions here: the R Syntax is just depending on the state of the options, not on the state of the engine.

Reply:
And the race condition is between the engine starting, then getting killed, then getting started, then getting killed again. Because if you change a bound value an engine will start running an analysis. 
But if another option changes the engine probably gets killed.

Anyway, the analysis already gets the options, so we can just have it return the syntax right?

You add a property to jaspResults that is called "rSyntax". Done.